packages/shared/src/errors/Error.ts

```diff
@@ -69,6 +90,58 @@ export class ClerkAPIResponseError extends Error {
   };
 }
 
+/**
+ * Custom error class for representing Clerk runtime errors.
+ *
+ * @class ClerkRuntimeError
+ * @example
+ *   throw new ClerkRuntimeError('An error occurred', { code: 'password_invalid' });
+ */
+export class ClerkRuntimeError extends Error {
+  /**
+   * Indicates that this is a Clerk runtime error.
+   *
+   * @type {boolean}
+   * @memberof ClerkRuntimeError
+   */
+  clerkRuntimeError: true;
+
+  /**
+   * The error message.
+   *
+   * @type {string}
+   * @memberof ClerkRuntimeError
+   */
+  message: string;
+
+  /**
+   * A unique code identifying the error, used for localization
+   *
+   * @type {string}
+   * @memberof ClerkRuntimeError
+   */
+  code: string;
+  constructor(message: string, { code }: { code: string }) {
+    super(message);
+
+    Object.setPrototypeOf(this, ClerkRuntimeError.prototype);
+
+    this.code = code;
+    this.message = message;
+    this.clerkRuntimeError = true;
+  }
+
+  /**
+   * Returns a string representation of the error.
+   *
+   * @returns {string} A formatted string with the error name and message.
+   * @memberof ClerkRuntimeError
+   */
+  public toString = () => {
+    return `[${this.name}]\nMessage:${this.message}`;
```

Comment:
❓ If the `toString()` is used to show the localized error message to the end-user I think we should probably need to remove the `this.name` prefix and show the localized message. Otherwise, shouldn't we also add the `code` ?

Reply:
`toString` is not used to display a localized message.

This class cannot have access to our localization mechanism that exist in `clerk-js` as this would introduce a cyclic dependency.

`translateError` inside our UI layer is in charge of localizing.

We are only interested in localizating the errors displayed in the UI, not the ones we are logging to console.